frontend/src/static/js/v_summary.js

```diff
@@ -98,16 +98,27 @@ window.vSummary = {
     },
 
     filterGroupSelection() {
+      this.getFiltered();
+
       // merge group is not allowed when group by none
+      // also reset merged groups
       if (this.filterGroupSelection === 'groupByNone') {
-        this.isMergeGroup = false;
+        this.mergedGroupsMap = {};
```

Comment:
seems like the `mergedGroupsMap` will be modified once every time `filterGroupSelection` is changed. Which means that `getFiltered` will be called twice (extra time when `mergeGroupsMap` watcher is called).

Reply:
I suppose that is ok? Because you need to call `getFiltered` the first time to reset the map and populate it with repo names, and the second time to merge repos.

Reply:
still I feel we should try to avoid call gieFiltered twice though since this involves lots of merging and sorting methods. if only the name is needed maybe can just extract the logic out here?

Reply:
> still I feel we should try to avoid call gieFiltered twice though since this involves lots of merging and sorting methods. if only the name is needed maybe can just extract the logic out here?

I have refactored this part. Now it will only do filtering and sorting (since sorting includes grouping by repo/author) but not do merging.